authors/apps/articles/models.py

```diff
@@ -48,7 +52,50 @@ def _get_unique_slug(self):
             num += 1
         return unique_slug
 
+    def get_likes_or_dislkes(**kwargs):
+        """
+        Fetches the number of likes and dislikes
+
+        args:
+            Keyword arguments:
+            These include the model,
+            id of an article and
+            the like status of the article
+
+        Returns:
+            filtered_likes(int): number of likes or dislikes of an article
+        """
+        likes = kwargs.get('model').objects.all().filter(
+            like_article=kwargs.get('like_article')
+        )
+        filtered_likes = likes.filter(article_id=kwargs.get('article_id'))
+        return filtered_likes.count()
+
     def save(self, *args, **kwargs):
         if not self.slug:
             self.slug = self._get_unique_slug()
         super(Article, self).save(*args, **kwargs)
+
+    @property
+    def likes_count(self):
+        return get_likes_or_dislkes(
+            model=ArticleLikes,
+            like_article=True,
+            article_id=self.pk
+        )
+
+    @property
+    def dislikes_count(self):
+        return get_likes_or_dislkes(
+            model=ArticleLikes,
+            like_article=False,
+            article_id=self.pk
+        )
+
+
+class ArticleLikes(models.Model):
+    article = models.ForeignKey(
+        Article, on_delete=models.CASCADE,
+        null=True, related_name="article_likes", blank=True)
+    like_article = models.BooleanField()
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
```

Comment:
@kimbugp there is a check for that such that if u like an article more than once, you are notified and the like count doesn't increase. If you u dislike after liking then the `like` is removed and the `dislike` is added. Otherwise when attempt to to make the user and article unique together, one can only carry out one particular action which defeats what is required from feature implementation.